### PR TITLE
Update the request object with sensible defaults and access methods, #13

### PR DESCRIPTION
**Pull request by diegoc-am**

### CODE REVIEW

File: examples/Example.java [hunk 1]
```diff
@@ -18,19 +18,15 @@ public static void main(String[] args) throws IOException {
 
     Request request = new Request();
     request.baseUri = "api.sendgrid.com";
-    Map<String,String> requestHeaders = new HashMap<String, String>();
-    requestHeaders.put("Authorization", "Bearer " + System.getenv("SENDGRID_API_KEY"));
-    request.headers = requestHeaders;
+    request.getHeaders().put("Authorization", "Bearer " + System.getenv("SENDGRID_API_KEY"));
```
Comment (thinkingserious) [inline examples/Example.java]: Can we use a setter here instead?

e.g. `request.addHeader("Authorization", "Bearer " + System.getenv("SENDGRID_API_KEY"));`

File: examples/Example.java [hunk 2]
```diff
@@ -18,19 +18,15 @@ public static void main(String[] args) throws IOException {
...
 
     // GET Collection
     request.method = Method.GET;
     request.endpoint = "/v3/api_keys";
-    Map<String,String> queryParams = new HashMap<String, String>();
-    queryParams.put("limit", "100");
-    queryParams.put("offset", "0");
-    request.queryParams = queryParams;
+    request.getQueryParams().put("limit", "100");
+    request.getQueryParams().put("offset", "0");
```
Comment (thinkingserious) [inline examples/Example.java]: Can we use a setter here instead?

e.g. `request.addQueryParam("offset", "0");`

File: examples/Example.java [hunk 3]
```diff
@@ -39,7 +35,7 @@ public static void main(String[] args) throws IOException {
     } catch (IOException ex) {
       throw ex;
     }
-    request.queryParams = null;
+    request.getQueryParams().clear();
```
Comment (thinkingserious) [inline examples/Example.java]: How about `request.clearQueryParams();`

**Review by thinkingserious — CHANGES_REQUESTED** [review]

**Comment by diegoc-am** [comment]
Will update it later tonight! Thanks for the feedback

File: examples/Example.java (hunk 1 above)
Comment (diegoc-am) [reply examples/Example.java]: fixed

File: examples/Example.java (hunk 2 above)
Comment (diegoc-am) [reply examples/Example.java]: fixed

File: examples/Example.java (hunk 3 above)
Comment (diegoc-am) [reply examples/Example.java]: fixed

**Comment by diegoc-am** [comment]
fixed!

**Review by thinkingserious — APPROVED** [review]

**Pull request merged**